integrations/gitlab/gitlab_integration/gitlab_service.py

```diff
@@ -169,21 +169,32 @@ def get_root_groups(self) -> List[Group]:
         return typing.cast(
             List[Group], [group for group in groups if group.parent_id is None]
         )
+    
+    def filter_groups_by_paths(self, groups_full_paths) -> List[Group]:
+        groups = self.gitlab_client.groups.list()
+        return typing.cast(
+            List[Group], [group for group in groups if group.attributes["full_path"] in groups_full_paths]
+        )
 
-    def create_webhooks(self) -> list[int | str]:
-        root_partial_groups = self.get_root_groups()
-        logger.info("Getting all the root groups to create webhooks for")
-        # Filter out root groups that are not in the group mapping and creating webhooks for the rest
+    def create_webhooks(self, groups_full_paths) -> list[int | str]:
+        if groups_full_paths:
+            logger.info("Getting all the specified groups in the mapping to create their webhooks")
+            partial_groups = self.filter_groups_by_paths(groups_full_paths)
+        else:
+            logger.info("Getting all the root groups to create their webhooks")
+            partial_groups = self.get_root_groups()
+        
+        # Filter out groups that are not in the group mapping and creating webhooks for the rest
         filtered_partial_groups = [
             group
-            for group in root_partial_groups
+            for group in partial_groups
             if any(
                 does_pattern_apply(mapping.split("/")[0], group.attributes["full_path"])
                 for mapping in self.group_mapping
             )
         ]
```

Comment:
I think the `create_webhooks` should only get a list of groups and do the creation, and move the logic of the group list to create in the function calling it

Reply:
i like that, on it

Reply:
done